core/src/main/kotlin/ch/ergon/dope/resolvable/clause/model/SetClause.kt

```diff
@@ -2,40 +2,51 @@ package ch.ergon.dope.resolvable.clause.model
 
 import ch.ergon.dope.DopeQuery
 import ch.ergon.dope.resolvable.clause.IUpdateSetClause
-import ch.ergon.dope.resolvable.clause.IUpdateUseKeysClause
 import ch.ergon.dope.resolvable.expression.TypeExpression
 import ch.ergon.dope.resolvable.expression.unaliased.type.Field
 import ch.ergon.dope.resolvable.formatToQueryString
 import ch.ergon.dope.validtype.ValidType
 
-class SetClause(
-    private val fieldAssignment: Pair<Field<out ValidType>, TypeExpression<out ValidType>>,
-    private vararg val fieldAssignments: Pair<Field<out ValidType>, TypeExpression<out ValidType>>,
-    private val parentClause: IUpdateUseKeysClause,
+class SetClause<T : ValidType>(
+    field: Field<T>,
+    value: TypeExpression<T>,
+    private val setAssignments: MutableList<SetAssignment<out ValidType>> = mutableListOf(),
+    private val parentClause: IUpdateSetClause,
 ) : IUpdateSetClause {
-    private fun Pair<Field<out ValidType>, TypeExpression<out ValidType>>.toDopeQuery(): DopeQuery {
-        val fieldDopeQuery = first.toDopeQuery()
-        val valueDopeQuery = second.toDopeQuery()
-        return DopeQuery(
-            queryString = "${fieldDopeQuery.queryString} = ${valueDopeQuery.queryString}",
-            parameters = fieldDopeQuery.parameters + valueDopeQuery.parameters,
-        )
+
+    init {
+        setAssignments.add(SetAssignment(field, value))
     }
 
     override fun toDopeQuery(): DopeQuery {
         val parentClauseDopeQuery = parentClause.toDopeQuery()
-        val fieldAssignmentDopeQuery = fieldAssignment.toDopeQuery()
-        val fieldAssignmentsDopeQuery = fieldAssignments.map { it.toDopeQuery() }
+        val setAssignmentsDopeQuery = setAssignments.map { it.toDopeQuery() }
         return DopeQuery(
             queryString = formatToQueryString(
                 "${parentClauseDopeQuery.queryString} SET",
-                fieldAssignmentDopeQuery.queryString,
-                *fieldAssignmentsDopeQuery.map { it.queryString }.toTypedArray(),
+                *setAssignmentsDopeQuery.map { it.queryString }.toTypedArray(),
             ),
-            parameters = fieldAssignmentsDopeQuery.fold(fieldAssignmentDopeQuery.parameters) {
-                    setExpressionParameters, field ->
-                setExpressionParameters + field.parameters
-            } + parentClauseDopeQuery.parameters,
+            parameters = setAssignmentsDopeQuery.fold(parentClauseDopeQuery.parameters) {
+                    setAssignmentsParameters, parent ->
+                setAssignmentsParameters + parent.parameters
+            },
+        )
+    }
+
+    fun <T : ValidType> set(field: Field<T>, value: TypeExpression<T>) =
+        SetClause(field, value, setAssignments = this.setAssignments, parentClause = this.parentClause)
```

Comment:
if you use my approach from above you can just write:
```
fun <T : ValidType> set(field: Field<T>, value: TypeExpression<T>) =
    SetClause(setAssignment, *setAssignments, SetAssignment(field, value), parentClause = this.parentClause)
```
(I would add the new assignment to the end of the list)

Reply:
As discussed offline, now the order is true to the order of calling the functions, done